### PR TITLE
Fix: Update kube-proxy using a documented version in AWS docs

### LINKED ISSUE #2536: kube-proxy update is using a non-existing image tag
**What happened?**
After upgrading a cluster from EKS 1.16 to 1.17 (platform version eks.2), I updated kube-proxy with `eksctl utils update-kube-proxy --approve`.
The DaemonSet has been updated with image tag v1.17.9 (following the master version). However, this image tag does not exist in AWS ECR repository for EKS.
```
docker pull 602401143452.dkr.ecr.us-east-1.amazonaws.com/eks/kube-proxy:v1.17.9
Error response from daemon: manifest for 602401143452.dkr.ecr.us-east-1.amazonaws.com/eks/kube-proxy:v1.17.9 not found: manifest unknown: Requested image not found
```
Only v1.17.6 (released with EKS 1.17 eks.1) and v1.17.7 (released with EKS 1.17 eks.2) are available.

**What you expected to happen?**
I expect that kube-proxy version is using a version made available by AWS (AWS doc https://docs.aws.amazon.com/fr_fr/eks/latest/userguide/update-cluster.html is reporting that kube-proxy version is actually v1.17.7 for EKS 1.17 cluster)

**How to reproduce it?**
Create an EKS cluster on 1.17 eks.2.
Execute `eksctl utils update-kube-proxy --approve`.

**Anything else we need to know?**
AWS documentation says that EKS 1.17 eks.2 is using Kubernetes 1.17.9 (https://docs.aws.amazon.com/eks/latest/userguide/platform-versions.html) but they are using 1.17.7 for kube-proxy (https://docs.aws.amazon.com/fr_fr/eks/latest/userguide/update-cluster.html).
eksctl kube-proxy update is broken as eksctl is using k8s version used by the control plane as image tag for kube-proxy

**Versions**
Please paste in the output of these commands:
```
$ eksctl version
0.25.0
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"17+", GitVersion:"v1.17.9-eks-4c6976", GitCommit:"4c6976793196d70bc5cd29d56ce5440c9473648e", GitTreeState:"clean", BuildDate:"2020-07-17T19:00:19Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"17+", GitVersion:"v1.17.9-eks-4c6976", GitCommit:"4c6976793196d70bc5cd29d56ce5440c9473648e", GitTreeState:"clean", BuildDate:"2020-07-17T18:46:04Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
```

**Logs**
```
$ cat cluster.yml | eksctl utils update-kube-proxy -f - --approve
[ℹ]  eksctl version 0.25.0
[ℹ]  using region eu-west-1
[ℹ]  "kube-proxy" is now up-to-date

$ kubectl -n kube-system get ds kube-proxy -o yaml
apiVersion: apps/v1                                  
kind: DaemonSet
metadata:
    spec:
      containers:
      - command:
        - kube-proxy
        - --v=2
        - --config=/var/lib/kube-proxy-config/config
        image: 602401143452.dkr.ecr.eu-west-1.amazonaws.com/eks/kube-proxy:v1.17.9
        imagePullPolicy: IfNotPresent
        name: kube-proxy
```

### PR DESCRIPTION
**Pull request by javiertoledos**
Some kubernetes control plane versions don't have a corresponding kube-proxy image tag with the same version. This change uses major and minor version to lookup for a documented version if exists, otherwise it uses the control plane version to build the image tag. The only consideration is to make sure to update the mapping each time AWS releases a new documented version.

Fixes #2536

### Description

As described in issue #2536, using only the control plane version to determine the correct image tag for kube-proxy when updated may lead to non-existing image. This causes some issues issues like worker nodes not being ready because CNI is failing to load due daemonset not having pods in the new worker nodes.

### Checklist
- [x] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [x] Manually tested
- [ ] Added labels for change area (e.g. `area/nodegroup`), target version (e.g. `version/0.12.0`) and kind (e.g. `kind/improvement`)
- [x] Make sure the title of the PR is a good description that can go into the release notes

### CODE REVIEW

File: pkg/addons/default/kube_proxy.go [hunk 1]
```diff
@@ -97,5 +98,21 @@ func UpdateKubeProxyImageTag(clientSet kubernetes.Interface, controlPlaneVersion
 }
 
 func kubeProxyImageTag(controlPlaneVersion string) string {
-	return fmt.Sprintf("v%s-eksbuild.1", controlPlaneVersion)
+	// See https://docs.aws.amazon.com/eks/latest/userguide/update-cluster.html for latest supported versions
+	documentedVersions := map[string]string{
```
Comment (michaelbeaumont) [inline pkg/addons/default/kube_proxy.go:102]: This can be a package level value

Comment (javiertoledos) [reply pkg/addons/default/kube_proxy.go]: OK, moved

File: pkg/addons/default/kube_proxy.go [hunk 2]
```diff
@@ -20,6 +21,15 @@ const (
 	KubeProxy = "kube-proxy"
 )
 
+// KubeProxyDocumentedVersions maps major and minor control plane version to latest documented AWS kube-proxy version
+// see https://docs.aws.amazon.com/eks/latest/userguide/update-cluster.html for latest supported versions
+var KubeProxyDocumentedVersions = map[string]string{
+	"1.17": "1.17.7",
```
Comment (michaelbeaumont) [inline pkg/addons/default/kube_proxy.go:27]: This is already out of date sadly enough :/

**Comment by michaelbeaumont** [comment]
@javiertoledos Before I merge a PR like this, I want to make sure there's not a better way. Please standby

**Comment by michaelbeaumont** [comment]
Closing this now as the issue is resolved and we'll need a different solution to ensure it doesn't happen again.

**Pull request closed without merging**